src/annotator/guest.js

```diff
@@ -146,44 +144,39 @@ export default class Guest extends Delegator {
       }
     });
 
-    this.plugins = {};
-
     /** @type {Anchor[]} */
     this.anchors = [];
 
+    // Setup the document type-specific integration consisting of metadata extraction,
+    // anchoring module and logic to respond to activity (eg. scrolling) in the page.
+    //
+    // In future the plan is to put these behind a common interface so that the
+    // `Guest` class doesn't need different code paths depending on the document type.
+
+    // nb. The `anchoring` field defaults to HTML anchoring and in PDFs is replaced
+    // by `PDFIntegration` below.
+    this.anchoring = anchoring;
+    this.documentMeta = new DocumentMeta(this.element);
+    if (config.documentType === 'pdf') {
+      this.pdfIntegration = new PDFIntegration(this.element, this);
+    }
+
     // Set the frame identifier if it's available.
     // The "top" guest instance will have this as null since it's in a top frame not a sub frame
     this.frameIdentifier = config.subFrameIdentifier || null;
 
-    this.anchoring = anchoring;
-
-    const cfOptions = {
+    this.crossframe = new CrossFrame(this.element, {
```

Comment:
> It misses a word to describe that it is an event emitter or a communication system.

I agree. I'd avoid using the term iframe specifically though because some of the frames (usually just the main frame) listening in won't be the main frame.